src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -315,24 +319,41 @@ public GetLatestLedgerResponse getLatestLedger() throws IOException, SorobanRpcE
    * @param transaction The transaction to simulate. It should include exactly one operation, which
    *     must be one of {@link InvokeHostFunctionOperation}, {@link ExtendFootprintTTLOperation}, or
    *     {@link RestoreFootprintOperation}. Any provided footprint will be ignored.
+   * @param resourceLeeway Additional resource include in the simulation.
    * @return A {@link SimulateTransactionResponse} object containing the cost, footprint,
    *     result/auth requirements (if applicable), and error of the transaction.
    * @throws IOException If the request could not be executed due to cancellation, a connectivity
    *     problem or timeout. Because networks can fail during an exchange, it is possible that the
    *     remote server accepted the request before the failure.
    * @throws SorobanRpcErrorResponse If the Soroban-RPC instance returns an error response.
    */
-  public SimulateTransactionResponse simulateTransaction(Transaction transaction)
+  public SimulateTransactionResponse simulateTransaction(
+      Transaction transaction, @Nullable ResourceLeeway resourceLeeway)
```

Comment:
is there a reason not to pass ResourceConfig here instead for more scalability? if more request attributes come up later, don't have to change this method signature again.

Reply:
Of course, I have made the modifications.